### PR TITLE
fix Issue 17430 - [scope] delegate can escape context ptr

### PR DESCRIPTION
**Pull request by WalterBright**
This works for me, so I added a test case for it.

### CODE REVIEW

File: test/fail_compilation/retscope2.d [hunk 1]
```diff
@@ -162,3 +162,67 @@ void foo800()
...
+
+
+
+
+
+
+
+
+
+
```
Comment (mathias-lang-sociomantic) [inline test/fail_compilation/retscope2.d]: There's no point in using this hack if you use `#line`.

Comment (WalterBright) [inline test/fail_compilation/retscope2.d]: The point of this is that there are 3 PRs, and I'll have to do a rebase for two of them. These blank lines will make that easier. The test cases already use `#line`.

Comment (rainers) [inline test/fail_compilation/retscope2.d]: I guess it won't help avoid merge conflicts (you could insert the different test cases in separate places instead), but if it makes life easier for you, let's go ahead.

**Review by rainers — APPROVED** [review]

**Comment by WalterBright** [comment]
The merge conflicts will still be there, they'll just be easier to resolve. Just delete the blank lines.

File: test/fail_compilation/retscope2.d (hunk 1 above)
Comment (MartinNowak) [inline test/fail_compilation/retscope2.d]: Git's text-diff does merge changes to the same file without conflict if there is at least one common line between the changes, so adding newlines helps, but adding more than a couple of newlines seems excessive.

**Pull request merged**